Framework/PythonInterface/mantid/utils/absorptioncorrutils.py

```diff
@@ -27,12 +33,137 @@ def _getBasename(filename):
     return name
 
 
-def calculate_absorption_correction(filename, abs_method, props, sample_formula, mass_density,
-                                    number_density=Property.EMPTY_DBL,
-                                    container_shape="PAC06",
-                                    num_wl_bins=1000,
-                                    element_size=1,
-                                    metaws=None):
+def _getCacheName(wkspname, cache_dir, abs_method):
+    """
+    Generate a MDF5 string based on given key properties.
+
+    :param wkspname: donor workspace containing absorption correction info
+    :param cache_dir: location to store the cached absorption correction
+    
+    return fileName(full path), sha1
+    """
+
+    # fix up the workspace name
+    prefix = wkspname.replace('__', '')
+
+    # parse algorithm used for absorption calculation
+    # NOTE: the acutal algorithm used depends on selected abs_method, therefore
+    #       we are embedding the algorithm name into the SHA1 so that we can
+    #       differentiate them for caching purpose
+    alg_name = {
+        "SampleOnly": "AbsorptionCorrection",
+        "SampleAndContainer": "AbsorptionCorrection",
+        "FullPaalmanPings": "PaalmanPingsAbsorptionCorrection",
+    }[abs_method]
+
+    # key property to generate the HASH
+    ws = mtd[wkspname]
+    # NOTE:
+    #  - The query for height is tied to a beamline, which is not a good design as it
+    #    will break for other beamlines
+    property_string = [
+        f"{key}={val}" for key, val in {
+            'wavelength_min': ws.readX(0).min(),
+            'wavelength_max': ws.readX(0).max(),
+            "num_wl_bins": len(ws.readX(0)) - 1,
+            "sample_formula": ws.run()['SampleFormula'].lastValue().strip(),
+            "mass_density": ws.run()['SampleDensity'].lastValue(),
+            "height_unit": ws.run()['BL11A:CS:ITEMS:HeightInContainerUnits'].lastValue(),
```

Comment:
Is this function supposed to work only for POWGEN? If so, there should be a warning or an exception thrown off when the beamline is not POWGEN

Reply:
Based on my understanding, this entire file is hard coded for POWGEN (see func `create_absorption_input` for more examples).
Given that the related user story is related to only POWGEN, I think we can safely assume that this whole script, `absorptioncorrutil.py` is intended for POWGEN only. :thinking:

Reply:
This is really ugly :disappointed: 
I can live with the technical debt, though. If users of other beamlines try the absorption correction in the future they'll be greeted with an exception. We'll cross that bridge then.